Using no_std on embedded

I'm trying to use this awesome library for an embedded project (just inference) but I'm having problems compiling it.

```bash
error[E0432]: unresolved import `__alloc::sync`
   --> ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/no-std-compat-0.4.1/src/generated.rs:173:48
    |
173 |     #[cfg(feature = "alloc")] pub use __alloc::sync::*;
    |                                                ^^^^ could not find `sync` in `__alloc`

For more information about this error, try `rustc --explain E0432`.
error: could not compile `no-std-compat` (lib) due to previous error
warning: build failed, waiting for other jobs to finish...
error[E0463]: can't find crate for `std`
  --> ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/num-traits-0.2.15/src/lib.rs:21:1
   |
21 | extern crate std;
   | ^^^^^^^^^^^^^^^^^ can't find crate
   |
   = note: the `thumbv6m-none-eabi` target may not support the standard library
   = help: consider building the standard library from source with `cargo build -Zbuild-std`

For more information about this error, try `rustc --explain E0463`.
```

I tried to run `cargo build -Zbuild-std` as suggested by I still get the error:

```bash
error[E0432]: unresolved import `alloc::sync`
 --> ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/gimli-0.26.2/src/read/dwarf.rs:2:12
  |
2 | use alloc::sync::Arc;
  |            ^^^^ could not find `sync` in `alloc`
```

It's probably a misconfiguration from my part but I was wondering if you have any suggestions. Thanks!

@antimora (I saw you in some of the `no_std` related issues :smile:)

This is the final error I'm getting with the fixes for the target:

```rust
error[E0432]: unresolved import `__alloc::sync`
   --> C:\Users\clowm\.cargo\registry\src\github.com-1ecc6299db9ec823\no-std-compat-0.4.1\src\generated.rs:173:48
    |
173 |     #[cfg(feature = "alloc")] pub use __alloc::sync::*;
    |                                                ^^^^ could not find `sync` in `__alloc`
```

Which is with `no-std-compat` 🙄

This is with an empty project with Cargo.toml:
```toml
[package]
name = "no-std-test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
dfdx = { path = "../dfdx", default-features = false, features = ["no-std"]}
```

and .cargo/config:
```
[target.thumbv6m-none-eabi]
rustflags = ["--cfg", "portable_atomic_unsafe_assume_single_core"]
```

Command to build is:
```
cargo build --target thumbv6m-none-eabi
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -25,9 +25,9 @@ features = ["nightly", "numpy", "safetensors", "cuda", "ci-check"]
 
 [dependencies]
 no-std-compat = { version = "0.4.1", default-features = false, features = [ "alloc", "compat_hash" ], optional = true }
-spin = { version = "0.9.6", default-features = false, features = ["spin_mutex", "rwlock"], optional = true }
+spin = { version = "0.9.8", default-features = false, features = ["spin_mutex", "rwlock", "portable_atomic"], optional = true }
 rand = { version = "0.8.5", default-features = false, features = ["std_rng"] }
-rand_distr = { version = "0.4.3", default-features = false, features = ["std_math"] }
+rand_distr = { version = "0.4.3", default-features = false }
 matrixmultiply = { version = "0.3.2", default-features = false, optional = true }
 zip = { version = "0.6.2", default-features = false, optional = true }
 cblas-sys = { version = "0.1.4", default-features = false, optional = true }
@@ -49,9 +49,9 @@ glob = { version = "0.3.1", optional = true }
 default = ["std", "fast-alloc", "cpu-par-matmul"]
 nightly = []
 
-std = ["cudarc?/std", "matrixmultiply?/std"]
+std = ["cudarc?/std", "matrixmultiply?/std", "rand_distr/std_math"]
 fast-alloc = ["std"]
-no-std = ["no-std-compat", "dep:spin", "cudarc?/no-std"]
+no-std = ["no-std-compat", "dep:spin", "cudarc?/no-std", "num-traits/libm"]
 
 cpu-seq-matmul = ["dep:matrixmultiply"]
 cpu-par-matmul = ["std", "dep:matrixmultiply", "matrixmultiply?/threading"]

